lua/lvim/plugin-loader.lua

```diff
@@ -3,174 +3,178 @@ local plugin_loader = {}
 local Log = require "lvim.core.log"
 local join_paths = utils.join_paths
-local in_headless = #vim.api.nvim_list_uis() == 0
 
--- we need to reuse this outside of init()
-local compile_path = join_paths(get_config_dir(), "plugin", "packer_compiled.lua")
-local snapshot_path = join_paths(get_cache_dir(), "snapshots")
-local default_snapshot = join_paths(get_lvim_base_dir(), "snapshots", "default.json")
+local plugins_dir = join_paths(get_runtime_dir(), "lazy", "plugins")
+
+local function remove_rtp_paths()
+  if os.getenv "LUNARVIM_RUNTIME_DIR" then
+    vim.opt.rtp:remove(join_paths(vim.call("stdpath", "data"), "site"))
+    vim.opt.rtp:remove(join_paths(vim.call("stdpath", "data"), "site", "after"))
+    vim.opt.rtp:remove(vim.call("stdpath", "config"))
+    vim.opt.rtp:remove(join_paths(vim.call("stdpath", "config"), "after"))
+  end
+end
 
 function plugin_loader.init(opts)
   opts = opts or {}
 
-  local install_path = opts.install_path
-    or join_paths(vim.fn.stdpath "data", "site", "pack", "packer", "start", "packer.nvim")
+  local lazy_install_dir = opts.install_path or join_paths(vim.fn.stdpath "data", "lazy", "plugins", "lazy.nvim")
 
-  local max_jobs = 100
-  if vim.fn.has "mac" == 1 then
-    max_jobs = 50
-  end
-
-  local init_opts = {
-    package_root = opts.package_root or join_paths(vim.fn.stdpath "data", "site", "pack"),
-    compile_path = compile_path,
-    snapshot_path = snapshot_path,
-    max_jobs = max_jobs,
-    log = { level = "warn" },
-    git = {
-      clone_timeout = 120,
-    },
-    display = {
-      open_fn = function()
-        return require("packer.util").float { border = "rounded" }
-      end,
-    },
-  }
-
-  if in_headless then
-    init_opts.display = nil
-    init_opts.git.clone_timeout = 300
-  end
-
-  if not utils.is_directory(install_path) then
+  if not utils.is_directory(lazy_install_dir) then
     print "Initializing first time setup"
-    print "Installing packer"
-    print(vim.fn.system { "git", "clone", "--depth", "1", "https://github.com/wbthomason/packer.nvim", install_path })
-    vim.cmd "packadd packer.nvim"
+    vim.fn.system {
+      "git",
+      "clone",
+      "--filter=blob:none",
+      "--single-branch",
+      "https://github.com/folke/lazy.nvim.git",
+      lazy_install_dir,
+    }
   end
 
-  local status_ok, packer = pcall(require, "packer")
-  if status_ok then
-    packer.on_complete = vim.schedule_wrap(function()
-      require("lvim.utils.hooks").run_on_packer_complete()
-    end)
-    packer.init(init_opts)
-  end
-end
+  remove_rtp_paths()
 
--- packer expects a space separated list
-local function pcall_packer_command(cmd, kwargs)
-  local status_ok, msg = pcall(function()
-    require("packer")[cmd](unpack(kwargs or {}))
+  vim.opt.runtimepath:prepend(lazy_install_dir)
+
+  -- Add plugins to rtp (needed for config:init)
+  -- TODO: is there a better way to do this?
+  require("lazy.core.util").ls(plugins_dir, function(path, name, type)
```

Comment:
What are you trying to achieve here? should not be needed?

Reply:
this adds the ability to `require` plugins before loading lazy.nvim, we need it currently because user's config is loaded before loading lazy.nvim, so that the user can change plugins's configuration and requiring plugins is needed to to that in some cases. It may not be needed in the future though, we might switch to lazy-requiring the plugins